aws/resource_aws_eks_addon.go

```diff
@@ -0,0 +1,332 @@
+package aws
+
+import (
+	"context"
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"log"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/eks"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/terraform-providers/terraform-provider-aws/aws/internal/keyvaluetags"
+)
+
+func resourceAwsEksAddon() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceAwsEksAddonCreate,
+		ReadContext:   resourceAwsEksAddonRead,
+		UpdateContext: resourceAwsEksAddonUpdate,
+		DeleteContext: resourceAwsEksAddonDelete,
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(5 * time.Minute),
+			Update: schema.DefaultTimeout(5 * time.Minute),
+			Delete: schema.DefaultTimeout(5 * time.Minute),
+		},
+
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"addon_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"cluster_name": {
```

Comment:
I guess we need `client_request_token` attribute?

Reply:
Typically a `ClientRequestToken` API parameter should be automatically set in the input type via `ClientRequestToken: aws.String(resource.UniqueId()),`